esrally/driver/runner.py

```diff
@@ -2877,6 +2877,16 @@ async def __call__(self, es, params):
                 else:
                     raise e
 
+            except elasticsearch.exceptions.ConnectionTimeout as e:
+                if last_attempt or not retry_on_timeout:
+                    raise e
```

Comment:
I understand that you're doing this for consistency, but for what it's worth this is equivalent to just `raise`